bottlecap/src/telemetry/events.rs

```diff
@@ -0,0 +1,349 @@
+use chrono::{DateTime, Utc};
+
+use serde::Deserialize;
+
+/// Payload received from the Telemetry API
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct TelemetryEvent {
+    /// Time when the telemetry was generated
+    pub time: DateTime<Utc>,
+    /// Telemetry record entry
+    #[serde(flatten)] // TODO: Figure out if this is ideal for our use case.
+    pub record: TelemetryRecord,
+}
+
+/// Record in a LambdaTelemetry entry
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(tag = "type", content = "record", rename_all = "lowercase")]
+pub enum TelemetryRecord {
+    /// Function log records
+    Function(String),
+
+    /// Extension log records
+    Extension(String),
+
+    /// Platform init start record
+    #[serde(rename = "platform.initStart", rename_all = "camelCase")]
+    PlatformInitStart {
+        /// Type of initialization
+        initialization_type: InitType,
+        /// Phase of initialisation
+        phase: InitPhase,
+        /// Lambda runtime version
+        #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
do you need this?

Reply:
The Skip serializing? Maybe not, because it should happen by default, right?

Reply:
Removed!